Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,87 @@
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def is_domain_internal(domain: str, internal_domains: List[str]) -> bool:
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+
+def main():
+    args = demisto.args()
+    internal_domains_list_name = args.get("InternalDomainsListName", "InternalDomains")
+    domains_to_check = argToList(args.get("Domains", ()))
+
+    # Get the list of internal domains from the XSOAR list:
+    internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list_name})[0]['Contents']
+    if "Item not found" in internal_domains:
+        return_error(f"The list name {internal_domains_list_name} does not exist.")
+
+    # Split internal domains from XSOAR list to be a list:
+    if internal_domains:
+        try:
+            internal_domains = internal_domains.split("\n")
+        except Exception as ex:
+            return_error(
+                f"Could not parse the internal domains list. Please make sure that the list contains domain names,"
+                f" separated by new lines.\nThe exact error is: {ex}")
+    else:
+        return_results(f"No internal domains were found under {internal_domains_list_name}.")
```

Comment:
If the list is empty, don't you want to return that the domains are not internal?

Reply:
Nope. It is better to let them know that the list needs to be configured instead of making the assumption that they're external. They got to have at least one internal domain. If they don't want to configure it they should not be using the script.